test/Potato/Data/Text/Zipper2Spec.hs

```diff
@@ -109,3 +104,85 @@ spec =
                     ["blah blah blah"]
     home tz_0 `shouldBe` tz_1h
     end tz_0 `shouldBe` tz_1e
+
+  it "top examples" $ do
+    top (TextZipper [] "" [] "dd d" ["a","b"]) `shouldBe` 
+         TextZipper [] "" [] "dd d" ["a","b"]
+    top (TextZipper ["hello"] "" [] "dd d" ["a","b"]) `shouldBe` 
+         TextZipper [] "" [] "hello" ["dd d", "a","b"]
+    top (TextZipper (reverse ["hello", "ok", "very good"]) "" [] "dd d" ["a","b"]) `shouldBe` 
+         TextZipper [] "" [] "hello" ["ok", "very good","dd d", "a","b"]
+    top (TextZipper (reverse ["hello", "ok", "very good"]) "" [] "dd d" ["a","b"]) `shouldBe` 
+         TextZipper [] "" [] "hello" ["ok", "very good","dd d", "a","b"]
+    top (TextZipper (reverse ["hello", "ok", "very good"]) "" ["a b c"] "dd d" ["a","b"]) `shouldBe` 
+         TextZipper [] "" [] "hello" ["ok", "very good","a b cdd d", "a","b"]
+    top (TextZipper (reverse ["hello", "ok", "very good"]) "" ["a b c", "d e "] "dd d" ["a","b"]) `shouldBe` 
+         TextZipper [] "" [] "hello" ["ok", "very good", "a b c", "d e dd d", "a","b"]
+
+  it "up examples" $ do
+    up (TextZipper [] "ab " [] "dd d" ["a","b"]) `shouldBe` 
+        (TextZipper [] "" [] "ab dd d" ["a","b"])
+    up (TextZipper ["cat dog"] "ab " [] "dd d" ["a","b"]) `shouldBe` 
+        (TextZipper [] "cat" [] " dog" ["ab dd d", "a","b"])
+    up (TextZipper ["cat dog"] "abfeeeere " [] "dd d" ["a","b"]) `shouldBe` 
+        (TextZipper [] "cat dog" [] "" ["abfeeeere dd d", "a","b"])
+    up (TextZipper (reverse ["cat dog", "bird_sky"]) "ab " [] "dd d" ["a","b"]) `shouldBe` 
+        (TextZipper ["cat dog"] "bir" [] "d_sky" ["ab dd d", "a","b"])
+    up (TextZipper [] "ab " ["story"] "dd d" ["a","b"]) `shouldBe` 
+        (TextZipper [] "" [] "ab storydd d" ["a","b"])
+    up (TextZipper [] "ab " ["story", "cool"] "dd d" ["a","b"]) `shouldBe` 
+        (TextZipper [] "" [] "ab story" ["cooldd d", "a","b"])
+    up (TextZipper ["a", "b"] "ab " ["story", "cool"] "dd d" ["a","b"]) `shouldBe` 
+        (TextZipper ["a", "b"] "" [] "ab story" ["cooldd d", "a","b"])
```

Comment:
shouldn't this be 

`TextZipper ["a"] "b" [] "ab story" ["cooldd d", "a","b"]`